hedgehog/src/Hedgehog/Internal/Gen.hs

```diff
@@ -522,6 +524,16 @@ instance (MonadGen m, Monoid w) => MonadGen (Strict.RWST r w s m) where
 ------------------------------------------------------------------------
 -- GenT instances
 
+-- technically, the Monoid constraint on @a@ could be weakened
```

Comment:
This comment  does not currently show up in haddock docs,  but I  can make it show up there since that might actually be  mildly useful information for users of the library.

Reply:
Can just leave this unannotated after adopting [George's feedback](https://github.com/hedgehogqa/haskell-hedgehog/pull/156/files#r167455259)